pinot-core/src/main/java/org/apache/pinot/core/realtime/impl/kafka/KafkaStreamConfigProperties.java

```diff
@@ -43,9 +43,12 @@
   public static class LowLevelConsumer {
     public static final String KAFKA_BROKER_LIST = "kafka.broker.list";
     public static final String KAFKA_BUFFER_SIZE = "kafka.buffer.size";
-    public static final int KAFKA_BUFFER_SIZE_DEFAULT = 512000;
+    public static final int KAFKA_BUFFER_SIZE_DEFAULT = 5120000;
```

Comment:
I can see that increasing buffer size will enable to fetch larger message. Other than that, what is the impact of increasing buffer? I guess that it will increase the throughput but may impact on latency?

Reply:
So in kafka consumer, there are 3 toggles to control the [consumer fetch behavior](https://kafka.apache.org/090/documentation.html#consumerconfigs):
1. fetch.min.bytes: this control when the kafka broker returns the data. This will impact the server latency (but usually minimal, like sub-seconds level) because kafka broker will try to ensure it has that many data before it return to the consumer. We are still using old config for this value.
2. fetch.message.max.bytes: this control how much data kafka broker supposes to return when it has more data. Usually, higher value can increase throughput when we have large lag (more efficient batching) and might have slight impact on consumer fetch latency (sub-second level) for sending more data if there are.
3. there is also a timeout config in stream config (fetch.timeout.millis) which allow us to control how much latency we will wait at most for kafka consumer. This will allow us to put an upper bound to consumer fetch latency if it is a concern.

Overall increasing this default value will increase latency at a minimal level (as it is bound by fetch.timeout.millis and kafka supposed to return data when reaching minBytes) and if this value is lower than kafka message size the consumer will fail silently at that message and get stuck. It would seem to be a small price to pay for getting pinot more flexible at handling large message and better catch-up performance

Reply:
@mcvsubbu I have reverted the default back to the old value.